Branch003

AllegraSalvatore opened the pull request:
Ecco una serie di modifiche che in locale mi restituiscono " code has been rated at 10.00/10"

Commit:
feat: modifiche sulle righe problematiche per pylint

Commit:
Modifiche per approvazione pylint più token e pylint passed local

AllegraSalvatore committed:
Update requirements_dev.txt

AllegraSalvatore committed:
Update ci.yml

AllegraSalvatore committed:
Update ci.yml

simone989 commented:
Ciao, ti consiglio di usare sempre Conventional Commits così da avere uniformità nella git history.

simone989 commented on src/bot_quake.py:
```diff
@@ -146,15 +158,16 @@ async def handle_message(update, context):
 def main() -> None:
-    TOKEN = os.environ["TELEGRAM_BOT"]
+    """ Funzione principale del bot"""
+    token_bot = os.environ["TELEGRAM_BOT"]
```
dato che il token non cambia mai ti consiglio di definire il nome della variabile in maiuscolo, come fatto prima.

AllegraSalvatore replied:
Il linter mi consigliava di usare lo snake_case altrimenti me lo scegnalava come errore. Che faccio cambio?

simone989 replied:
Ok nessun problema, ignora il mio consiglio.
Probabilmente te lo segnala come errore perchè il linter non comprende che quella è una variabile in sola lettura.

AllegraSalvatore replied:
Ok, grazie comunque!

linuchs commented:
> Ciao, ti consiglio di usare sempre Conventional Commits così da avere uniformità nella git history.

D'accordo grazie!

simone989 commented on src/bot_quake.py:
```diff
@@ -68,18 +67,22 @@ async def file_reader(update, context):
+        f"&maxdepth=1000&minlat={zona.minlat}&maxlat={zona.maxlat}"
+        f"&minlon={zona.minlon}&maxlon={zona.maxlon}"
+        f"&minversion=100&orderby=time-asc&format=text&limit=100"
+    )
```
Che ve ne pare di definire una funzione separata che preso in input i parametri ritorna l'url già pronto da chiamare ?

Così da rendere più chiaro cosa si sta facendo in questa sezione del codice.

AllegraSalvatore replied:
Fatto!

simone989 commented on src/bot_quake.py:
```diff
@@ -110,18 +114,23 @@ async def file_reader(update, context):
-        """Benvenuto in BOTQUAKE questo è un sistema automatizzato per visualizzare l'ultimo evento sismico tra gli eventi degli ultimi 7 giorni in una zona di interesse intorno al vulcano Etna.
+async def start(update: Update, context: ContextTypes.DEFAULT_TYPE)-> None:
+    """ Invia una descrizione del bot"""
+    print(context.args)
```
rimuovete i `print(context.args)` nel codice se non sono più necessari

simone989 commented on src/bot_quake.py:
```diff
@@ -1,13 +1,17 @@
-# la funzione imposta i valori temporali dei dati da scaricare in un range che va dal giorno attuale indietro di un valore d_range
 def get_date_range(d_range):
+    """ la funzione imposta i valori temporali dei dati da scaricare
+    in un range che va dal giorno attuale indietro di un valore d_range"""
```
Spostate gesta funziona in un nuovo file utils/helper

AllegraSalvatore replied:
Fatto!

Commit:
Se applicato, questo commit crea il file
gethelp.py nella nova cartella helper/gethelp,
il file contiene la funzione get_date_range
precedentemente in botquake.py

Commit:
Aggiunge la funzione generate_url e utils/helpers

Risolve parte delle richieste in: #10, relativamente
allo spostare la funzione get_date_range in un nuovo file utils/helper
e la craazione di una funzione che sarà chiamata generate_url che
ritorna l'url già pronto

Commit:
Merge branch 'branch003' of https://github.com/linuchs/Botquake into branch003
